lib/mad_cart/store/base.rb

```diff
@@ -42,6 +42,41 @@ def init_connection
         @connection ||= execute_delegate(klass.connection_delegate, @init_args)
       end
 
+      def validate_by_path(path)
+        check_for_errors do
+          connection.get(path)
+        end
+        return true
```

Comment:
This method gets called by the `valid?` method in the store implementation. Perhaps renaming this method to `valid_by_path?` would make more sense?